### PR TITLE
Added reencrypt_workspace CLI

### LINKED ISSUE #2708: Add reencrypt_workspace CLI

### PR DESCRIPTION
**Pull request by Max-7**
Closes #2708

### CODE REVIEW

File: parsec/core/cli/reencrypt_workspace.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Parsec Cloud (https://parsec.cloud) Copyright (c) AGPLv3 2016-2021 Scille SAS
+
+import click
+
+
```
Comment (touilleMan) [inline parsec/core/cli/reencrypt_workspace.py:5]: single newline between groups of imports

File: parsec/core/cli/reencrypt_workspace.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+async def _reencrypt_workspace(config: CoreConfig, device: LocalDevice, name: EntryName) -> None:
+    async with logged_core_factory(config, device) as core:
+        workspace = core.find_workspace_from_name(name)
+        workspace_id = workspace.id
+        workspace_fs = core.user_fs.get_workspace(workspace_id)
+        reenc_needs = await workspace_fs.get_reencryption_need()
+        if not reenc_needs.need_reencryption:
+            raise RuntimeError("The workspace does not need to be reencrypted")
```
Comment (touilleMan) [inline parsec/core/cli/reencrypt_workspace.py:22]: Raising an error for this seems a bit to harsh, instead I'd go with a nice `click.echo` message then return (so status code of the process would be 0)

**Review by touilleMan — CHANGES_REQUESTED** [review]

File: parsec/core/cli/reencrypt_workspace.py (hunk 1 above)
Comment (Max-7) [reply parsec/core/cli/reencrypt_workspace.py]: A bit surprised this wasn't caught by the pre-commit, but it's fixed.

File: parsec/core/cli/reencrypt_workspace.py (hunk 2 above)
Comment (Max-7) [reply parsec/core/cli/reencrypt_workspace.py]: Done.

**Review by touilleMan — APPROVED** [review]
:+1:

**Pull request merged**